### PR TITLE
Add .pick and .assign methods to Runnable

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

**Comment by eyurtsev** [comment]
Is there a way to pick and rename?

**Review by baskaryan — COMMENTED** [review]
are we sure
 ```python
x.assign(foo=bar).pick(["foo", "baz"]) | y  | z
```
 is clearer than 
 ```python
x | RunnableAssign(foo=bar) | RunnablePick(["foo", "baz"]) | y | z
```
? kinda nice only having one chaining operator (`|`) imo. but if we're going the route of having instance methods that are also chaining operators should we also add `.append()` and `.extend()` so that if someone wants they can just use that syntax?
```python
x.assign(foo=bar).pick(["foo", "baz"]).extend([y, z])
```

File: libs/core/langchain_core/runnables/base.py [hunk 1]
```diff
@@ -396,6 +396,30 @@ def __ror__(
         """Compose this runnable with another object to create a RunnableSequence."""
         return RunnableSequence(first=coerce_to_runnable(other), last=self)
 
+    def pick(self, keys: Union[str, List[str]]) -> Runnable[Any, Any]:
```
Comment (baskaryan) [inline libs/core/langchain_core/runnables/base.py:399]: what if this was `*keys: str`?

**Comment by baskaryan** [comment]
> Is there a way to pick and rename?

maybe better to add a separate `.rename(foo="bar", baz="buz", ...)` and let user chain pick and rename themselves?

File: libs/core/langchain_core/runnables/base.py (hunk 1 above)
Comment (nfcampos) [reply libs/core/langchain_core/runnables/base.py:399]: A single key gets you the value itself whereas a list of key(s) gets you a dict with those keys

Comment (baskaryan) [reply libs/core/langchain_core/runnables/base.py:399]: could do `(self, *keys: str, key: Optional[str]=None)`?

Comment (baskaryan) [reply libs/core/langchain_core/runnables/base.py:399]: separately should we support `name: Optional[str]=None` and pass that to RunnableAssign?

Comment (baskaryan) [reply libs/core/langchain_core/runnables/base.py:399]: or just pass through **kwargs

**Pull request merged**